src/main/java/org/openjdk/jextract/impl/Utils.java

```diff
@@ -318,6 +319,17 @@ static String quote(char ch) {
         }
     }
 
+    static String layoutClassNameInDeclaration(MemoryLayout layout) {
+        if (layout.getClass().getName().contains(".internal")) {
```

Comment:
Could the check be based on interface vs class itself rather than ".internal" in package name of internal class?